### PR TITLE
Backend for PJNZ upload

### PR DESCRIPTION
**Pull request by hillalex**
2 endpoints, 1 for uploading a PJNZ file (for now just saving to disk and returning some dummy json) and one for retrieving metadata (for now just reading from disk). The implementations of both will change once the R API is there.

The front-end code will follow in a separate PR

### CODE REVIEW

File: app/src/main/kotlin/org/imperial/mrc/hint/AppProperties.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@Configuration
+class ConfiguredAppProperties: AppProperties {
+
+    override val uploadDirectory = properties["upload_dir"].toString()
+
+    companion object
+    {
+        val properties = Properties().apply {
+            load(getResource("config.properties").openStream())
+            val global = File("/etc/orderly/web/config.properties")
```
Comment (EmmaLRussell) [inline app/src/main/kotlin/org/imperial/mrc/hint/AppProperties.kt]: Should be /etc/hint..
Would this class be more testable if the path orproperties could be injected?

File: app/src/main/kotlin/org/imperial/mrc/hint/HintApplication.kt [hunk 2]
```diff
@@ -2,6 +2,7 @@ package org.imperial.mrc.hint
 
 import org.springframework.boot.SpringApplication
 import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.context.annotation.ComponentScan
```
Comment (EmmaLRussell) [inline app/src/main/kotlin/org/imperial/mrc/hint/HintApplication.kt]: Is this import needed?

File: app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.web.multipart.MultipartFile
+import java.io.File
+
+@RestController
+@RequestMapping("/baseline")
+class BaselineController(val context: WebContext,
+                         val pac4jConfig: Config,
+                         val appProperties: AppProperties) {
+
+    @PostMapping("/pjnz/upload")
```
Comment (EmmaLRussell) [inline app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt]: Doesn't really matter but I suppose the /upload isn't really needed as it's implied by a POST to a pjnz endpoint?

File: app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+        // TODO request serialised data for this id from the R API
+        // for now just read basic file info from upload dir
+        val pjnzFiles = File("${appProperties.uploadDirectory}/$id/pjnz")
+                .listFiles()
+
+        return if (pjnzFiles != null && pjnzFiles.any()) {
+            val fileName = pjnzFiles.first().name
+            val countryName = fileName.split("_").first()
+            "{\"pjnz\": { \"filename\": \"$fileName\", \"country\": \"$countryName\"}}"
```
Comment (EmmaLRussell) [inline app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt]: I take it we're eventually going to take the usual approach of serializing objects to JSON?

Comment (hillalex) [reply app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt]: In this case we'll actually not be doing any serialising, we'll just be getting JSON from the R API and passing it straight through

File: app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt (hunk 3 above)
Comment (hillalex) [reply app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt]: Good point, nicer to be RESTful where we can!

File: app/src/main/kotlin/org/imperial/mrc/hint/AppProperties.kt (hunk 1 above)
Comment (hillalex) [reply app/src/main/kotlin/org/imperial/mrc/hint/AppProperties.kt]: Not sure what you mean by this class being testable?

Comment (hillalex) [reply app/src/main/kotlin/org/imperial/mrc/hint/AppProperties.kt]: Ah you mean to catch typos of this kind? Sure

File: src/gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
```
Comment (EmmaLRussell) [inline src/gradle/wrapper/gradle-wrapper.properties]: Did you mean to add this src dir on this branch?

Comment (hillalex) [reply src/gradle/wrapper/gradle-wrapper.properties]: :woman_facepalming:

**Pull request merged**